lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -651,12 +644,12 @@ def do_3d_projection(self, renderer):
             zvec = np.array([[0], [0], [self._sort_zpos], [1]])
             ztrans = proj3d.proj_transform_vec(zvec, renderer.M)
             return ztrans[2][0]
-        elif tzs.size > 0 :
+        elif tzs.size > 0:
             # FIXME: Some results still don't look quite right.
             #        In particular, examine contourf3d_demo2.py
             #        with az = -54 and elev = -45.
             return np.min(tzs)
-        else :
+        else:
```

Comment:
~~~
else:
    return np.min(vzs) if vzs.size else np.nan
~~~

Reply:
I see where you're coming from but mixing in a ternary in a if... elif chain looks meh to me.

Reply:
Actually, I don't think that the three blocks are logically equal. To me, the current form

~~~
if self._sort_zpos is not None:
    [use _sort_zpos]
elif tzs.size > 0:
    [use min(tzs)]
else:
    [use fallback]
~~~
is logically more like

~~~
if self._sort_zpos is not None:
    [use _sort_zpos]
else:
    if tzs.size > 0:
        [use min(tzs)]
    else:
        [use fallback]
~~~

Usually `tzs` are expected to be there and `nan` is only a fallback if they are not. Thus it would be preferable to

~~~
if self._sort_zpos is not None:
    [use _sort_zpos]
else:
    [use min(tzs) with fallback]
~~~

However, I agree that depends on the point of view you take. And I won't argue further if you still prefer to leave it as is.

Reply:
I know (hence the "I see where you're coming from" above), but it still looks not so nice to me.  (Really, best would be `min(tzs, default=np.nan)` but np.min doesn't have that and builtins.min is likely slower.)